packages/codegen/src/codegen.ts

```diff
@@ -161,45 +164,79 @@ export async function athenaCodegen({
         },
       ],
     })),
-    ...(await preset.buildGeneratesSection({
-      baseOutputDir: baseDir,
-      config: {
-        useTypeImports: true,
-      },
-      presetConfig: {
-        baseTypesPath: path.relative(baseDir, schemaTypesOutputPath),
-        folder: '__generated',
-        extension,
-      },
-      schema: parsedSchema,
-      documents: gqlTagDocuments,
-      pluginMap: {
-        typescriptOperations: typescriptOperationsPlugin,
-        typedDocumentNode: typedDocumentNodePlugin,
-      },
-      plugins: [
-        {
-          typescriptOperations: {
-            nonOptionalTypename: true,
+    ...(production
```

Comment:
Can we break this up into a standalone object that gets declared and passed to `buildGeneratesSection`?

e.g. something like:
```ts
const presetConfig = { /* whatever the shared config is */ }
if (production) {
  presetConfig.someProdThing = blah
}
await preset.buildGeneratesSection(presetConfig);
```

It looks like the production and non-production configs are almost identical, so just duplicating the whole thing makes it hard to tell what is actually changing here.

Reply:
good call.